improve docs and samples for glossaries and custom models

mohamedshabanofficial opened the pull request:
solves #17918

kristapratico commented on sdk/translation/azure-ai-translation-document/README.md:
```diff
@@ -290,6 +290,119 @@ To see how to use the Document Translation client library with Azure Storage Blo
+The following section provides some insights for some of the advanced translation features such as glossaries and custom translation models.
+
+### Glossaries
+Glossaries are domain-specific dictionaries. For example, if you want to translate some medical-related documents, you may need support for the many words, terminology, and idioms in the medical field which you can't find in the standard translation dictionary or you simply need specific translation. This is why Document Translator provides support for glossaries. Document Translator supports glo ...
```
love the example here!

kristapratico commented on sdk/translation/azure-ai-translation-document/README.md:
````diff
@@ -290,6 +290,119 @@ To see how to use the Document Translation client library with Azure Storage Blo
+You can read more about - for example - tsv files [here][tsv_files].\
+In order to use glossaries with Document Translator, you first need to upload your glossaries file to a blob container, and then provide it's SaS url to Document Translator as in the following sample:
+
+```python
````
we can link out to the glossary sample instead of providing the code in the readme.

kristapratico commented on sdk/translation/azure-ai-translation-document/README.md:
````diff
@@ -290,6 +290,119 @@ To see how to use the Document Translation client library with Azure Storage Blo
+Instead of using Document Translator's engine for translation, you can use your own custom azure machine/deep learning model.
+For more info on how to create and provision your own custom azure translation model, please follow the instructions here: [Build, deploy, and use a custom model for translation][custom_translation_article]
+
+```python
````
same here, I'd link out to the sample you created in this PR

kristapratico commented on sdk/translation/azure-ai-translation-document/README.md:
```diff
@@ -290,6 +290,119 @@ To see how to use the Document Translation client library with Azure Storage Blo
+|Tab Separated Values/TAB|.tsv/.tab|A tab-delimited raw-data file used by spreadsheet programs.
+
+You can read more about - for example - tsv files [here][tsv_files].\
+In order to use glossaries with Document Translator, you first need to upload your glossaries file to a blob container, and then provide it's SaS url to Document Translator as in the following sample:
```
I think we need to include some information on how to format the glossary file. From reading this, it's not immediately clear to me what I should put in the file.

mohamedshabanofficial commented:
@kristapratico 
i added some sample file for glossaries .tsv (same as .tab)
you can find the PR [here](https://github.com/Azure/azure-sdk-for-python/pull/19545)
added some random medical word translation (to be consistent with the readme)
is it better if i asked the service team to provide some sample glossary files?

kristapratico commented on sdk/translation/azure-ai-translation-document/README.md:
```diff
@@ -323,6 +323,40 @@ To see how to use the Document Translation client library with Azure Storage Blo
+The following section provides some insights for some of the advanced translation features such as glossaries and custom translation models.
+
+### **Glossaries**
+Glossaries are domain-specific dictionaries. For example, if you want to translate some medical-related documents, you may need support for the many words, terminology, and idioms in the medical field which you can't find in the standard translation dictionary or you simply need specific translation. This is why Document Translator provides support for glossaries. 
```
I think we should stick to calling the service Document Translation like we do in other places. (this comment applies generally)

```suggestion
Glossaries are domain-specific dictionaries. For example, if you want to translate some medical-related documents, you may need support for the many words, terminology, and idioms in the medical field which you can't find in the standard translation dictionary, or you simply need specific translation. This is why Document Translation provides support for glossaries. 
```

kristapratico commented on sdk/translation/azure-ai-translation-document/README.md:
```diff
@@ -323,6 +323,40 @@ To see how to use the Document Translation client library with Azure Storage Blo
+
+Document Translator supports glossaries in the following formats:
+
+|**File Type**|**Extension**|**Description**|
```
it looks like they also support csv files for glossaries: https://docs.microsoft.com/en-us/azure/cognitive-services/translator/document-translation/overview#supported-glossary-formats

kristapratico commented on sdk/translation/azure-ai-translation-document/README.md:
```diff
@@ -323,6 +323,40 @@ To see how to use the Document Translation client library with Azure Storage Blo
+, and you can also see a sample file [here][sample_tsv_file].
+
+#### **How Use Glossaries in Document Translator**
+In order to use glossaries with Document Translator, you first need to upload your glossaries file to a blob container, and then provide it's SaS url to Document Translator as in the code samples [sample_translation_with_glossaries.py][sample_translation_with_glossaries].
```
I think we should make sure we be specific and say the SAS URL to the glossary file so people don't mistaken it for SAS URL to the container.

kristapratico commented on sdk/translation/azure-ai-translation-document/README.md:
```diff
@@ -323,6 +323,40 @@ To see how to use the Document Translation client library with Azure Storage Blo
+Instead of using Document Translator's engine for translation, you can use your own custom azure machine/deep learning model.
+
+#### **How To Create a Custom Translation Model**
+For more info on how to create and provision your own custom azure translation model, please follow the instructions here: [Build, deploy, and use a custom model for translation][custom_translation_article]
```
azure should be capitalized
```suggestion
For more info on how to create and provision your own custom Azure translation model, please follow the instructions here: [Build, deploy, and use a custom model for translation][custom_translation_article]
```

kristapratico commented on sdk/translation/azure-ai-translation-document/README.md:
```diff
@@ -323,6 +323,40 @@ To see how to use the Document Translation client library with Azure Storage Blo
+
+#### **How To Use a Custom Translation Model With Document Translator**
+In order to use a custom translation model with Document Translator, you first 
+need to create and deploy your model, then follow the code sample [sample_translation_with_custom_model.py][sample_translation_with_custom_model]. to use with Document translator.
```
nit: unexpected period

```suggestion
need to create and deploy your model, then follow the code sample [sample_translation_with_custom_model.py][sample_translation_with_custom_model] to use with Document translator.
```

kristapratico commented on sdk/translation/azure-ai-translation-document/samples/async_samples/sample_translation_with_custom_model_async.py:
```diff
@@ -0,0 +1,91 @@
+            )
+
+    async with client:
+        job = await client.create_translation_job(inputs=[inputs])  # type: JobStatusResult
```
samples need to be updated now that we have `begin_translation`  :)

kristapratico commented on sdk/translation/azure-ai-translation-document/README.md:
```diff
@@ -436,6 +470,12 @@ This project has adopted the [Microsoft Open Source Code of Conduct][code_of_con
+[sample_translation_with_custom_model_async]: https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/translation/azure-ai-translation-document/samples/async_samples/sample_translation_with_custom_model_async.py
+
+[custom_translation_article]: https://docs.microsoft.com/azure/cognitive-services/translator/custom-translator/quickstart-build-deploy-custom-model
+[tsv_files_wikipedia]: https://en.wikipedia.org/wiki/Tab-separated_values
```
```suggestion
[tsv_files_wikipedia]: https://wikipedia.org/wiki/Tab-separated_values
```

kristapratico commented:
> @kristapratico
> i added some sample file for glossaries .tsv (same as .tab)
> you can find the PR [here](https://github.com/Azure/azure-sdk-for-python/pull/19545)
> added some random medical word translation (to be consistent with the readme)
> is it better if i asked the service team to provide some sample glossary files?

The glossary sample file we provide should reflect translation changes if somebody tries to use it. If you run translation using the glossary file you created, does the output show that the glossary was used? If so, I think we should use your sample. If not, then let's check with the service team to see if they have any good examples.

kristapratico commented on sdk/translation/azure-ai-translation-document/README.md:
```diff
@@ -323,6 +323,38 @@ To see how to use the Document Translation client library with Azure Storage Blo
+
+Document Translation supports glossaries in the following formats:
+
+|**File Type**|**Extension**|**Description**|**Samples**|
+|---------------|---------------|---------------|---------------|
+|Tab-Separated Values/TAB|.tsv, .tab|Read more on [wikipedia][tsv_files_wikipedia]|[glossary_sample.tsv][sample_tsv_file]|
+|Comma-Seperated Values|.csv|Read more on [wikipedia][csv_files_wikipedia]|[glossary_sample.csv][sample_csv_file]|
+|Localization Interchange File Format|.xlf, .xliff|Read more on [wikipedia][xlf_files_wikipedia]|[glossary_sample.xlf][sample_xlf_file]|
```
let's just link to this table in the documentation since it's inevitable it will change as they add more formats: https://docs.microsoft.com/azure/cognitive-services/translator/document-translation/overview#supported-glossary-formats

I think we should still call out that there are glossary samples in the samples folder, though.

kristapratico commented on sdk/translation/azure-ai-translation-document/README.md:
```diff
@@ -323,6 +323,38 @@ To see how to use the Document Translation client library with Azure Storage Blo
+|Localization Interchange File Format|.xlf, .xliff|Read more on [wikipedia][xlf_files_wikipedia]|[glossary_sample.xlf][sample_xlf_file]|
+
+#### **How Use Glossaries in Document Translation**
+In order to use glossaries with Document Translation, you first need to upload your glossaries file to some blob container, and then provide the SaS url to of this blob container - which contains the glossary files - to Document Translation as in the code samples [sample_translation_with_glossaries.py][sample_translation_with_glossaries]. (Note: Don't conflate with blob container with the one you ...
```
IIRC, you'll need to provide the sas url of the glossary file, not the container it's in.

kristapratico commented on sdk/translation/azure-ai-translation-document/README.md:
```diff
@@ -342,7 +374,7 @@ See full SDK logging documentation with examples [here][sdk_logging_docs].
 
 Optional keyword arguments can be passed in at the client and per-operation level.
-The azure-core [reference documentation][azure_core_ref_docs]
+The Azure-Core [reference documentation][azure_core_ref_docs]
```
capitalization was correct before -  https://pypi.org/project/azure-core/

kristapratico commented on sdk/translation/azure-ai-translation-document/samples/async_samples/sample_translation_with_custom_model_async.py:
```diff
@@ -0,0 +1,90 @@
+FILE: sample_translation_with_custom_model_async.py
+
+DESCRIPTION:
+    This sample demonstrates how to create a translation job and apply custom azure translation model when doing the translation.
```
replace "job" with "operation"

kristapratico commented on sdk/translation/azure-ai-translation-document/samples/async_samples/sample_translation_with_custom_model_async.py:
```diff
@@ -0,0 +1,90 @@
+
+    client = DocumentTranslationClient(endpoint, AzureKeyCredential(key))
+
+    inputs = DocumentTranslationInput(
```
since we're only using one input, suggest using the "overloaded" begin_translation method and passing the `category_id` keyword argument.

kristapratico reviewed (approved):
LGTM
